datalad/metadata/metadata.py

```diff
@@ -497,7 +497,9 @@ def _get_metadata(ds, types, global_meta=None, content_meta=None, paths=None):
                     if isinstance(v, dict):
                         vset.add(ReadOnlyDict(v))
                     elif isinstance(v, list):
-                        vset.add(tuple(v))
+                        def rtuple(t):
+                            return tuple(map(rtuple, t)) if isinstance(t, list) else t
+                        vset.add(rtuple(v))
```

Comment:
I will give it a try, but imho the whole thing should be recursive, i.e. if there is a list of dicts etc.

Reply:
It is already recursive (for lists). If this works for you I'll merge the other conditional branches into this helper function too.